backend/gce/gce.go

```diff
@@ -0,0 +1,249 @@
+	if err != nil {
+		return nil, fmt.Errorf("error creating client: %v", err)
+	}
+
+	g.computeService, err = compute.New(client)
+	if err != nil {
+		return nil, fmt.Errorf("error creating compute service: %v", err)
+	}
+
+	networkName, err := networkFromMetadata()
+	if err != nil {
+		return nil, fmt.Errorf("error getting network metadata: %v", err)
+	}
+
+	g.project, err = projectFromMetadata()
+	if err != nil {
+		return nil, fmt.Errorf("error getting project: %v", err)
+	}
+
+	instanceName, err := instanceNameFromMetadata()
+	if err != nil {
+		return nil, fmt.Errorf("error getting instance name: %v", err)
+	}
+
+	instanceZone, err := instanceZoneFromMetadata()
+	if err != nil {
+		return nil, fmt.Errorf("error getting instance zone: %v", err)
+	}
+
+	g.gceNetwork, err = g.computeService.Networks.Get(g.project, networkName).Do()
+	if err != nil {
+		return nil, fmt.Errorf("error getting network from compute service: %v", err)
+	}
+
+	g.gceInstance, err = g.computeService.Instances.Get(g.project, instanceZone, instanceName).Do()
+	if err != nil {
+		return nil, fmt.Errorf("error getting instance from compute service: %v", err)
+	}
+
+	found, err := g.handleMatchingRoute(l.Subnet.String())
+	if err != nil {
+		return nil, fmt.Errorf("error handling matching route: %v", err)
+	}
+
+	if !found {
+		_, err := g.insertRoute(l.Subnet.String())
+		if err != nil {
+			return nil, fmt.Errorf("error inserting route: %v", err)
+		}
+	}
+
+	return &backend.SubnetDef{
+		Net: l.Subnet,
+		MTU: extIface.MTU,
+	}, nil
+}
+
+func (g *GCEBackend) Run() {
+	log.Info("GCE backend running")
+	subnet.LeaseRenewer(g.ctx, g.sm, g.network, g.lease)
+}
+
+func (g *GCEBackend) Stop() {
+	log.Info("GCE backend stopping")
+	g.cancel()
+}
+
+func (g *GCEBackend) Name() string {
+	return "gce"
+}
+
+//returns true if an exact matching rule is found
+func (g *GCEBackend) handleMatchingRoute(subnet string) (bool, error) {
+	matchingRoute, err := g.getRoute(subnet)
+	if err != nil {
+		if apiError, ok := err.(*googleapi.Error); ok {
+			if apiError.Code != 404 {
+				return false, fmt.Errorf("error getting the route err: %v", err)
+			}
+			return false, nil
+		}
+		return false, fmt.Errorf("error getting googleapi: %v", err)
+	}
+
+	if matchingRoute.NextHopInstance == g.gceInstance.SelfLink {
+		log.Info("Exact pre-existing route found")
+		return true, nil
+	}
+
+	log.Info("Deleting conflicting route")
+	operation, err := g.deleteRoute(subnet)
+	if err != nil {
+		return false, fmt.Errorf("error deleting conflicting route : %v", err)
+	}
+
+	for i := 0; ; i++ {
+		time.Sleep(time.Second)
+		operation, err := g.computeService.GlobalOperations.Get(g.project, operation.Name).Do()
+		if err != nil {
+			return false, fmt.Errorf("error fetching delete operation status: %v", err)
+		}
+
+		if i%5 == 0 {
+			log.Infof("Delete operation status: %v waiting for completion...", operation.Status)
+		}
+
+		if operation.Status == "DONE" {
+			break
+		}
+
+		if i > 100 {
```

Comment:
I would put the predicate into the `for` statement and then change
```
if operation.Status == "DONE" {
    return false, nil
}
```

and last return to `return false, fmt.Errorf("timeout waiting for delete operation to finish")`